### DISCUSSION TITLE
Migrating embedding model from text-embedding-ada-002 to text-embedding-3-small

### DISCUSSION THREAD

**Discussion by dolukhanov** [spring-projects/spring-ai#725] [Q&A]
My application was using text-embedding-ada-002 for its embedding model, using PgVector.

After the release of text-embedding-3-small, I wrote a migration script that took the contents of the vector_store table and regenerated the embedding for each content using text-embedding-3-small.

Now when running a similarity search (where the EmbeddingClient is set to the new embedding model), each query that previously generated results now generates 0 results.

Is there some configuration change with regards to dimensions or similar that needs to be changed to use the text-embedding-3-small embedding model?

**Comment by PiratePanda978** [comment]
same issue in my application, using redis db

**Comment by yantakus** [comment]
Hey @dolukhanov @PiratePanda978 could you please share how you managed to make it work? Having the same issue. Thanks.
